src/Atmos/Model/ref_state.jl

```diff
@@ -54,23 +55,35 @@ end
 vars_state(m::HydrostaticState, ::Auxiliary, FT) =
     @vars(ρ::FT, p::FT, T::FT, ρe::FT, ρq_tot::FT)
 
+atmos_init_ref_state_pressure!(m, _...) = nothing
+function atmos_init_ref_state_pressure!(
+    m::HydrostaticState{P, F},
+    atmos::AtmosModel,
+    aux::Vars,
+    geom::LocalGeometry,
+) where {P, F}
+    z = altitude(atmos, aux)
+    _, p = m.virtual_temperature_profile(atmos.param_set, z)
+    aux.ref_state.p = p
+end
 
 function atmos_init_aux!(
     m::HydrostaticState{P, F},
     atmos::AtmosModel,
     aux::Vars,
+    tmp::Vars,
     geom::LocalGeometry,
 ) where {P, F}
     z = altitude(atmos, aux)
     T_virt, p = m.virtual_temperature_profile(atmos.param_set, z)
     FT = eltype(aux)
     _R_d::FT = R_d(atmos.param_set)
+    k = vertical_unit_vector(atmos, aux)
+    ∇Φ = ∇gravitational_potential(atmos, aux)
 
-    # Replace density by computation from pressure
-    # ρ = -1/g*dpdz
-    ρ = p / (_R_d * T_virt)
+    # density computation from pressure ρ = -1/g*dpdz
+    ρ = -k' * tmp.∇p / (k' * ∇Φ)
     aux.ref_state.ρ = ρ
-    aux.ref_state.p = p
```

Comment:
Don't we still need pressure here?

Reply:
I don't think so. To take the gradient of pressure the reference state is initialized in two phases. The reference state pressure is initialized in `atmos_init_ref_state_pressure!`, before this function is called. Maybe it would be clearer to get `p` directly from `aux.ref_state` here ?

Reply:
Ah, I see, thanks!